### PR TITLE
Acquisition fastem: perform pre-calibrations outside the region of ac…

### PR DESCRIPTION
**Pull request by tepals**
…quisition

* perform the pre-calibrations on the field with index (-1, -1) to limit beam damage in the actual acquired area.
* do not skip test_get_pos_first_tile in the mock test class, because get_pos_first_tile does work with mocking.
* in the mocked test cases update multibeam.resolution.value from (800, 800) to (6400, 6400), to match the actual system.

Will test on Monday

### CODE REVIEW

**Comment by tepals** [comment]
Testing was successful, pre-calibration is done outside the ROA.

File: src/odemis/acq/fastem.py [hunk 1]
```diff
@@ -320,13 +320,18 @@ def run(self):
...
-        self.move_stage_to_next_tile()
-
         if self._pre_calibrate:
+            # Move the stage to the tile with index (-1, -1), to ensure the autofocus and image translation pre-align
+            # are done to the top left of the first field, outside the region of acquisition to limit beam damage.
+            self.field_idx = (-1, -1)
+            self.move_stage_to_next_tile()
             self.pre_calibrate()
 
+        # Move the stage to the first tile, to ensure the correct position is
```
Comment (Sabrina85) [inline src/odemis/acq/fastem.py:330]: Does it not make sense to move line 320/321 here but outside of the if statement? Now you move to the first tile of the ROA, then you move outside of the ROA for pre-calibration and then back to the first tile. Instead you could only move to the (-1, -1) directly and then to (0,0). That would save one step I think.

File: src/odemis/acq/test/fastem_test.py [hunk 2]
```diff
@@ -655,6 +655,18 @@ def test_get_abs_stage_movement_overlap(self):
             actual_position = task.get_abs_stage_movement()  # [m]
             numpy.testing.assert_allclose(actual_position, expected_position)
 
+            # Verify that for pre-calibrations compared to the top left corner of the ROA, the stage
```
Comment (Sabrina85) [inline src/odemis/acq/test/fastem_test.py:658]: I suggest to make it a separate test case. This test is already extensive and you can just make a simple test for only checking this and only for one overlap.

**Review by Sabrina85 — CHANGES_REQUESTED** [review]

File: src/odemis/acq/fastem.py (hunk 1 above)
Comment (tepals) [reply src/odemis/acq/fastem.py:330]: Line 321 does not move the stage, it just calculates the position of the first tile. To move to (-1, -1) the position of the first tile needs to have been calculated.

Comment (Sabrina85) [reply src/odemis/acq/fastem.py:330]: Ah right this was needed as `move_stage_to_next_tile` needs to know the position of the first tile.

File: src/odemis/acq/fastem.py [hunk 3]
```diff
@@ -320,12 +320,22 @@ def run(self):
...
 
-        # Move the stage to the first tile, to ensure the autofocus and image translation pre-align are
-        # done in the correct location and the correct position is stored in the megafield metadata yaml file.
-        self.move_stage_to_next_tile()
-
         if self._pre_calibrate:
+            # The pre-calibrations should be done a full field outside the ROA,
+            # therefore temporarily set the overlap to zero.
+            overlap_init = self._roa.overlap
+            self._roa.overlap = 0
```
Comment (tepals) [inline src/odemis/acq/fastem.py:327]: This is a bit ugly, but I don't have a better idea of solving this.

File: src/odemis/acq/fastem.py [hunk 4]
```diff
@@ -321,11 +321,16 @@ def run(self):
         self._pos_first_tile = self.get_pos_first_tile()
 
         if self._pre_calibrate:
+            # The pre-calibrations should be done a full field outside the ROA,
```
Comment (Sabrina85) [inline src/odemis/acq/fastem.py:324]: missing word "on" ?

File: src/odemis/acq/fastem.py (hunk 3 above)
Comment (Sabrina85) [reply src/odemis/acq/fastem.py:327]: mhm I also do not have a better idea right now.

**Review by Sabrina85 — CHANGES_REQUESTED** [review]

**Review by Sabrina85 — APPROVED** [review]

**Pull request merged**